### PR TITLE
Vagrant setup

### PR DESCRIPTION
**Pull request by ChiragGhanshani**
Created a vagrant configuration for simpler setup of development environment.

### CODE REVIEW

**Comment by zmughal** [comment]
You should rebase and squash the commits so that the removal is not there. Also, I have an example of using Vagrant in https://github.com/CougarCS/autobahn

**Comment by ChiragGhanshani** [comment]
I figured out what the issue was. I'll commit the fix as soon as the provisioning is done, but it might be a while because of all the tests perlbrew runs when installing a perl version.

**Comment by ChiragGhanshani** [comment]
@zmughal @StayerX I think it's ready. Merge it if you don't see any problems with it.

**Comment by StayerX** [comment]
Could it be possible to edit the README.md file with some information about how to set up and run Vagrant?

**Comment by StayerX** [comment]
The only thing that is missing is that it requires virtual-box and vagrant pre-installed.
I agree with this merge.

**Comment by StayerX** [comment]
Note: guys what is your opinion on something like this, that works differently in debian and redhat?

```
#!/bin/bash
# Check if the platform is debian/ubuntu based or Redhat/CenOS (no support for the rest.. yet)
python -mplatform |egrep -i "Ubuntu|Debian" && debianbased=true||debianbased=false

if [ "$debianbased" = true ] ; then
    echo -n 'Assuming Debian based GNU/Linux,\nProceding with installation'
    sudo xargs apt-get install \
        -y --no-install-recommends \
        < debian-packages-vagrant
else
    echo -n 'Assuming RedHat based distribution'
    sudo xargs yum install \
        -y --no-install-recommends \
        < redhat-packages-vagran
 fi
```

File: VagrantFiles/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  #   push.app = "YOUR_ATLAS_USERNAME/YOUR_APPLICATION_NAME"
+  # end
+
+  # Enable provisioning with a shell script. Additional provisioners such as
+  # Puppet, Chef, Ansible, Salt, and Docker are also available. Please see the
+  # documentation for more information about their specific syntax and use.
+   config.vm.provision "shell", path: "./provision.sh"
+   config.vm.provision :shell, privileged: false, path: "./perlbrewSetup.sh"
+   config.vm.provision :shell, privileged: false, path: "./cpanmSetup.sh"
+#     sudo apt-get update
```
Comment (zmughal) [inline VagrantFiles/Vagrantfile]: These comments can be cleaned out if the actual code is running in the scripts.

**Comment by zmughal** [comment]
@StayerX, I like that idea! I was thinking about doing something like that in another pull request. Could you copy that idea to another issue on this repository?

The pull request looks fine! Go ahead and merge (but see my next comment), @ChiragGhanshani.

There is something odd about the base box used for this Vagrant configuration:

```
==> default: Machine booted and ready!
==> default: Checking for guest additions in VM...
    default: No guest additions were detected on the base box for this VM! Guest
    default: additions are required for forwarded ports, shared folders, host only
    default: networking, and more. If SSH fails on this machine, please install
    default: the guest additions and repackage the box to continue.
    default:
    default: This is not an error message; everything may continue to work properly,
    default: in which case you may ignore this message.
```

The base box does _not_ have the VirtualBox guest additions installed, so syncing the shared folders is done the slow way using `rsync` rather than mounting using the `vboxsf` FS type. We should look for a better base box. I've used Ubuntu boxes with Vagrant in the past and they were setup correctly.

---

There are some style issues that I want to fix, but I'll first try to write a style guide on the wiki, so don't worry about that.

Also, we should write a developer project layout document. In the future, the project will start using many repositories at once, so keeping things consistent for all of us will help with sharing commands. I'll write that up later and make the changes here when I've done that.

**Comment by zmughal** [comment]
@ChiragGhanshani, in the commit message of https://github.com/project-renard/devops/pull/3/commits/d3a373a59f48ac9bd704448d0393ddf42c33415e, it says:

```
removed accidentally commited vagrant vm data from commit
```

You should remove that bit from the message since after squashing it doesn't apply. You can do that with an interactive rebase and select the `reword` action.

**Comment by ChiragGhanshani** [comment]
@zmughal I'm merging now, but I'll change the base box later on.

**Comment by zmughal** [comment]
Awesome! I  will assign you to that issue.

**Pull request merged**